### ISSUE TITLE
ReadSVG: Continuously Rasterize

### ISSUE THREAD

**Issue by devernay** [NatronGitHub/Natron#264]
To quote a Natron user:
"To be honest, the ONE thing--coming from a Motion Graphics slant--that I'd like to see is a way to import SVG into Natron and have it keep the edges of the vector sharp (what is known as "continuous rasterization" in another well-known package) during changes in animation."

I think this could be done by a simple modification of the https://github.com/NatronGitHub/openfx-arena/blob/master/Extra/ReadSVG.cpp plugin.

First all all, it would work only for perspective transformations, this includes those made by the CornerPin, Card3D, and Transform plugins. The Position plugin cannot be used, because, as explained in its documentation, "it does not concatenate transforms" (use Transform instead).

Let us say you want to apply a transform to a SVG file. First, create a Read node to read the SVG, then apply any combination of CornerPin, Card3D, and Transform downstream. The result should look OK geometrically, by "pixelized" since it was rasterized before applying the transforms.

Now, you would create another Read node that reads the *same* SVG, and connect its "Sync" input to the output of the last transform node. In the properties panel of this Read node, in the "Decoder options" section, there will be a new checkbox parameter, called "Apply Upstream Transform", and the output would be a resterization of the SVG *with* the transform applied to it, and thus "continuously rasterized".

How does that sound?

**Comment by splidje** [comment]
Two alternatives:
1) it seems that decoder options can be used to control the pixel output of a reader. More keyframeable options could be added which give access to SVG's translate, rotate, scale and shear. These could then be linked via expressions to other nodes. - This would still make motion graphics a little tricky, as you don't have the power of complex concatenations
2) A new node, which isn't a GenericReader, but is an "SVG Renderer" or something. It takes a path to an SVG and does exactly what you describe. Perhaps also with the feature of merging its render with the "background" piped into it.

**Comment by Songtech-0912** [comment]
@devernay Have worked on trying to implement vectors in Natron with a friend recently, and we came to this conclusion after a couple of months:

* Continous rasterization is probably only going to truly work if we can get OFX nodes to accept **vector data** rather than raster data as inputs, or if we create new nodes exclusively for working with vector data that do not use OFX (e.g. `vectorTransform`, `vectorMerge`, `vectorGrade`)
* If we are going to get **vector data** to be supported, we need a **complete rewrite** of the engine
* A **complete rewrite** is neither feasible nor achievable in the short-term and is similar on the scale of a full 3D implementation
* Many more people want to get a 3D workspace than vectors support, despite both being similar in terms of effort, and Natron's limited efforts should best be spent fixing bugs and achieving the [milestones](https://github.com/NatronGitHub/Natron/milestones) instead
* It would be best to work on vector graphics in a separate Qt-based app similar in architecture to Natron, so that when the time is ready, we can port the widgets and libraries we developed for this separate vector graphics app back into Natron (which is our current plan)

**Comment by devernay** [comment]
Yes, you would have to write a vector suite for OpenFX, separate from the Image suite. And same goes for 3D.

Vector graphics can be edited and animated in inkscape. Then fine adjustments can be made in Natron.

Honestly, I think users should stick to Blender for 3D: whatever we have, we won't be as powerful as Blender.

**Comment by Songtech-0912** [comment]
Yeah, so close this issue then? I think the matter is pretty much resolved, but no problem if you have something to add to the discussion.
